pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -325,6 +326,22 @@ func (w *Webhook) inject(pod *corev1.Pod, _ string, _ dynamic.Interface) error {
 	return w.injectAutoInstruConfig(pod, libsToInject, autoDetected, injectionType)
 }
 
+func injectSecurityClientLibraryProducts(pod *corev1.Pod) {
+	injectProductActivationKey(pod, "asm.enabled", "DD_APPSEC_ENABLED")
```

Comment:
nit: 

the function name here seems a bit vague to me. 
IMHO, the code would be more readable and easier to understand if we simply did something like this:

```
_ = mutatecommon.InjectEnv(pod, corev1.EnvVar{
				Name:	"DD_APPSEC_ENABLED",
				Value: 	config.GetBool("admission_controller.auto_instrumentation.asm.enabled"),
			});
}

_ = mutatecommon.InjectEnv(pod, corev1.EnvVar{
				Name:	"DD_IAST_ENABLED",
				Value: 	config.GetBool("admission_controller.auto_instrumentation.iast.enabled"),
			});
}

_ = mutatecommon.InjectEnv(pod, corev1.EnvVar{
				Name:	"DD_SCA_ENABLED",
				Value: 	config.GetBool("admission_controller.auto_instrumentation.sca.enabled"),
			});
}
```

WDYT?

Reply:
As previously mentioned, I think the current implementation fits better with the need for a 3 state config. I've tighten the naming to `injectEnvVarIfConfigKeySet` in 4406e81130

Reply:
I see

In that case, I suggest something like this:

```
if config.Datadog.IsSet("admission_controller.auto_instrumentation.asm.enabled") {
  _ = mutatecommon.InjectEnv(pod, corev1.EnvVar{
				  Name:	"DD_APPSEC_ENABLED",
				  Value: 	config.GetBool("admission_controller.auto_instrumentation.asm.enabled"),
			  });
  }
} // same for the others
```

But it is fine as you have it. I am just suggesting this because passing the config key as an argument to the function is not a pattern I am used to. But both can work.